Add contributing guide, and update readme (#144)
ustaxes/UsTaxes: .github/CONTRIBUTING.md

- [Pull Request Guidelines](#pull-request-guidelines)
- [Development Guide](#development-guide)

## Maintaining team

* [Aidan Grimshaw](http://github.com/thegrims)
* [Zak Patterson](http://github.com/zakpatterson)

## Issue Reporting Guidelines

- The issue list of this repo is exclusively for bug reports and feature requests. Please come to the discord with general questions you may have.

- If you have a question, you can get quick answers from the [Discord chat](https://discord.gg/dAaz472mPz).

- If your issue is resolved but still open, please close it. In case you found a solution by yourself, it could be helpful to explain how you fixed it.

- Generally, there are many features still remaining to produce a valid 1040 for most people. We hope you will view this project in light of your individual circumstances and expertise and contribute accordingly! For example if you have a certain type of income or receive a certain credit or deduction, and those are not yet implemented, we would gratefully assist you to implement them!

## Pull Request Guidelines

- Start all commits with a capital letter and imperitive mood, no period at the end.

> - Each commit first line should be max 50 characters, followed by a blank line, followed by as much detail as you think is needed, wrapped at 80 characters.

Author: The final squashed commit message does matter though, that's going to be the title of the PR followed by the bullets of each squashed commit, which the maintainer then has to sift.

- Github PR title style is the same as commit first line style.

> - It's OK to have multiple small commits as you work on the PR - we apply squash merges to all PRs at the end. In particular, you are free to rebase your commits so long as your pull request is in Draft status. As soon as you take it out of draft status, request a review, or receive a review on your pull request, please do not rebase or force push.

Author: Once the pr is ready for review we or another may pull the branch to test it out, and may want to contribute another assisting commit. So if the pr branch is rebased and force pushed in the meantime, then it will conflict with the other contributor's history. Does that make sense?

> - If adding new feature:

Reviewer: ```suggestion
- If adding a new feature please provide some reason to add this feature.
```

  - Provide some reason to add this feature.

- If fixing a bug:
  - If you are resolving a special issue, add `Fixes: #xxx` (#xxx is the issue id) in your PR body. When your PR is merged, the maintainer will move this information into the final commit message for our release log.
  - Provide detailed description of the bug in the PR, or link to an issue that does.

## Development Guide

### General Setup

First, [join our Discord server](https://discord.gg/dAaz472mPz) and let us know that you want to contribute. This way we can point you in the right direction and help ensure your contribution will be as helpful as possible.

To set up your machine for development, review the [Architecture doc](ARCHITECTURE.md), for required links to set up NPM > 7 and Rust.

Next, fork and clone this repo. 

Try 

* `npm run desktop` to try out the desktop application. Setting the environment variable `BROWSER=none` will stop the web browser from also loading for you.
* `npm run start` will serve the site locally on port 3000, so you can view it at `localhost:3000` in your browser.

#### Stack

> This app uses React and Redux on typescript. For this reason the structure of this app may look familiar to you if you have used redux before, and it may seem opaque and confusing to you if you have not used Redux before. [Here is a great guide you can refer to about redux](https://redux.js.org/tutorials/fundamentals/part-1-overview)

Reviewer: ```suggestion
This app uses Typescript React and Redux. The structure of this app may look familiar to you if you have used redux before, and it may seem opaque and confusing to you if you have not used Redux before. [Here is a great guide you can refer to about redux](https://redux.js.org/tutorials/fundamentals/part-1-overview)
```

> For our purposes, there are a few concepts I can summarize here. In order to manage state between many different components and concerns in a project, we dispatch actions that update a piece of state in a global state object. Each of these actions is received by a reducer that applies that change to the state variable. All of this logic is contained in [src/redux](src/redux).

Reviewer: ```suggestion
##### State Management
In order to manage the state of tax information between many different components and concerns in a project, we dispatch actions that update a piece of state in a global state object. Each of these actions is received by a reducer that applies that change to the global application state. All of this logic is contained in [src/redux](src/redux).
```

* `src/redux/actions.ts`: All the actions that can be sent to update our state
* `src/redux/data.ts`: No functionality, just type definitions for our global state variable.
* `src/redux/reducer.ts`: All logic for updating our global state. 

### Directories Overview

- [`src/components`](../src/components) Contains React forms and **all UI**.
- [`src/data`](../src/data) Contains static data such as a list of states and a list of tax brackets

> - [`src/irsForms`](../src/irsForms) These are typescript model implementations of the actual IRS pdfs. Each form provides the data to be filled into the final PDF via an array where each index must match the expected index in the PDF. We use the convention that methods are named after the actual line referred to in the PDF. So `const l1 = (): number = ...` will be the function to call to get the numeric value needed on line 1 of that form.

Reviewer: Should also probably add something about tax calculations occurring in this directory
